src/content/blog/definitelyformatted/index.mdx

```diff
@@ -0,0 +1,311 @@
+---
+description: "How we migrated the massive DefinitelyTyped repository from using a linter for formatting to the dprint formatter."
+image:
+  alt: ""
+  src: "~/assets/blog/default.webp"
+pubDate: 2024-04-29
+title: "DefinitelyFormatted"
+---
+
+import battleOfKöniggrätz from "~/assets/blog/battle-of-königgrätz.webp";
+import LabeledImage from "~/components/blog/mdx/LabeledImage.astro";
+
+Last year, I had the pleasure of leading a project to onboard the massive [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository to using a dedicated formatter, [dprint](https://dprint.dev), with some members of the TypeScript community and team.
+This change took us nearly half a year from start to end: initial discussions in late May and the last PR merged in early October.
+It went through some twists and turns but I'm happy with how it all turned out.
+
+> _"No plan survives first contact with the enemy"_
+> -Helmuth von Moltke
+
+## Context
+
+DefinitelyTyped is one of the largest, most active repositories on GitHub.
+It contains over 9,000 folders storing community-authored TypeScript type definitions for JavaScript projects that don't ship their own types.
```

Comment:
> over 9,000

Note from @sandersn:

> technically, 8851, although it might have been over 9000 in 2023

```plaintext
$ git checkout master
  ...
$ ls -l types | wc -l
    8787
$ git checkout 1374f05e833b7f2edd100e76d48041ba2759b814
  ...
$ ls -l types | wc -l
    8638
```

Grr, so much for the 9,001 reference. Thanks @sandersn!